### PR TITLE
Create imagedraw

### LINKED ISSUE #39: Add imagedraw
Investigate and add imagedraw.

https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html

**Acceptance criteria**

- Draw paths
- Class based (Class in the library called LLPlot with a method called get backplot)
- Works with example
- Set the background colour
- Automatically calculate image size
- Set G path color
- Set the line thickness

### PR DESCRIPTION
**Pull request by Brett-Parker**
fixes #39

### CODE REVIEW

**Comment by Brett-Parker** [comment]
@dubstar-04 I think this is an acceptable place to merge this PR. This meets the minimum requirements for #39. Draws paths, class based with LLPlot and it is imbedded with the example. 

I will move on to #50 to improve it and make it better.

File: LLPlot.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            y_end = self.imageSize[1] + (code[x]['x'] * scale) - 25
+
+            draw.line((x_start, y_start, x_end, y_end), fill=line_color, width=self.lineThickness)
+            i += 1
+            if x != (len(code) - 1):
+                x += 1
+
+        # Mirror because its draw flipped.
+        img = ImageOps.flip(img)
+        img.save(self.imageName + self.imageType)
```
Comment (dubstar-04) [inline LLPlot.py:166]: could you add the option to save to a specified path?

File: LLPlot.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        self.g2Color = (0, 256, 0)
+        self.g3Color = (0, 256, 0)
+        self.lineThickness = 2
+
+        self.__min_x = 500000
+        self.__min_y = 500000
+        self.__max_x = -500000
+        self.__max_y = -500000
+
+    def set_background_color(self, color):
```
Comment (dubstar-04) [inline LLPlot.py:22]: is color the america english and colour the correct english? :)

File: LLPlot.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        """"Set background color of image"""
+        if isinstance(color, tuple) and len(color) == 3:
+            try:
+                self.background = color
```
Comment (dubstar-04) [inline LLPlot.py:26]: can we add a issue to extend this to allow setting a transparent background and saving as a .png?

File: LLPlot.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            except Exception:
+                raise Warning('Unknown color! Color is RGB. For example (255, 255, 255)')
+
+    def set_path_color(self, g_number, color):
+        """Set the colour of g path. For example G0, (255,255,255)"""
```
Comment (dubstar-04) [inline LLPlot.py:31]: This is a nice feature!

File: LLPlot.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        draw = ImageDraw.Draw(img)
+
+        i = 0
+        x = 1
+        while i <= (len(code) - 1):
+            line_color = self.__get_line_color(code[i]['g'][0])
+
+            # / 2 to draw from the center of the image
+            # - 25 to offset the draw point from the edge of the image
+            x_start = (self.imageSize[0] / 2) + (code[i]['z'] * scale) - 25
```
Comment (dubstar-04) [inline LLPlot.py:154]: are only lines currently handled?

what about G2 / G3?

**Review by dubstar-04 — CHANGES_REQUESTED** [review]

File: LLPlot.py (hunk 1 above)
Comment (Brett-Parker) [reply LLPlot.py]: Added in commit 0933275

File: LLPlot.py (hunk 5 above)
Comment (Brett-Parker) [reply LLPlot.py:163]: Yes only lines at the moment. We will add to #50 the additional work. Need to understand what sort of things we can expect.

File: LLPlot.py (hunk 3 above)
Comment (Brett-Parker) [reply LLPlot.py:40]: Added to commit 07ab80c

File: LLPlot.py (hunk 2 above)
Comment (Brett-Parker) [reply LLPlot.py]: Added to commit 343fe27

**Comment by Brett-Parker** [comment]
@dubstar-04 Not going to do anymore on this now. Will wait for it to be merged.

**Pull request merged**